src/org/zaproxy/zap/view/JCheckBoxTree.java

```diff
@@ -84,7 +84,7 @@ void fireCheckChangeEvent(CheckChangeEvent evt) {
         Object[] listeners = listenerList.getListenerList();
         for (int i = 0; i < listeners.length; i++) {
             if (listeners[i] == CheckChangeEventListener.class) {
-                ((CheckChangeEventListener) listeners[i + 1]).checkStateChanged(evt);
```

Comment:
This is correct, what's wrong is the for (it should be e.g. `int i = listeners.length - 2; i >= 0; i -= 2`).